addons/l10n_us/models/res_partner_bank.py

```diff
@@ -2,17 +2,15 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import re
-from odoo import fields, models, api, _
+from odoo import models, api, _
 from odoo.exceptions import ValidationError
 
 
 class ResPartnerBank(models.Model):
     _inherit = ['res.partner.bank']
 
-    aba_routing = fields.Char(string="ABA/Routing", help="American Bankers Association Routing Number")
-
-    @api.constrains('aba_routing')
-    def _check_aba_routing(self):
+    @api.constrains('clearing_number')
+    def _check_clearing_number_us(self):
         for bank in self:
-            if bank.aba_routing and not re.match(r'^\d{1,9}$', bank.aba_routing):
+            if self.env.company.country_code == 'US' and bank.clearing_number and not re.match(r'^\d{1,9}$', bank.clearing_number):
```

Comment:
Shouldn't we use `bank.country_code == 'US'` instead of `self.env.company.country_code == 'US'`?

and maybe add `country_code` to `api.constrains`